src/Build/Logging/FancyLogger/FancyLoggerProjectNode.cs

```diff
@@ -116,6 +122,12 @@ public FancyLoggerTargetNode AddTarget(TargetStartedEventArgs args)
         {
             if (args.Importance != MessageImportance.High) return null;
             MessageCount++;
+            // Detect output messages using regex
+            // var match = Regex.Match(args.Message, $"(?<={args.ProjectFile} -> )(.*)");
+            var match = Regex.Match(args.Message!, $"(?<=.* -> )(.*)");
+            if (match.Success)
+                ProjectOutputExecutable = match.Value;
+
```

Comment:
This definitely accomplishes the goal! However, it's pretty inefficent on .NET Framework 4.8:

1. This method is https://referencesource.microsoft.com/#System/regex/system/text/regularexpressions/Regex.cs,798 which calls
2. https://referencesource.microsoft.com/#System/regex/system/text/regularexpressions/Regex.cs,9aef9195b829acff
3. Which creates a new `Regex` object for each call
4. That object's `Match` method is called, creating a `Match` object
5. The `match.Value` allocates a string to hold the latter half of the message.
6. The `Regex` and `Match` objects become garbage after this AddMessage call

This isn't quite as bad as it might be because `new Regex`  has a cache that will not require it to parse + compile the expression again, but it's still allocatey.

There are a couple of options to fix this. The easiest one is to move the regex to a `static` (so it's never garbage but there's only ever one), but that still allocates a `Match` object.

Since this regex is pretty simple I prefer a more manual but minimal-allocation approach:

```suggestion

            const string FinalOutputMarker = " -> ";
            var i = args.Message!.IndexOf(FinalOutputMarker);
            if (i > 0)
            {
                ProjectOutputExecutable = args.Message!.Substring(i + FinalOutputMarker.Length);
            }
```

The only allocation there is the substring, which could be avoided by preserving a span/memory into the original string, but this will allow the GC to collect the whole `BuildMessageEventArgs`, so I think it's better.

Reply:
Can there be a case in which a high priority message in this format but **without** the project name at the beginning exist?

Reply:
Yes, at minimum one could have a target that says `<Message Importance="High" Text="X -> Y" />`, which is basically all the message we're looking for here does:

https://github.com/dotnet/msbuild/blob/5a635cf1218a5d7836114de1e392e7613d569675/src/Tasks/Microsoft.Common.CurrentVersion.targets#L4715

You could either check the initial substring or check whether the parent Target is `CopyFilesToOutputDirectory`.

Reply:
Alright, so I should only check for messages whose parent target is `CopyFilesToOutputDirectory` using `IndexOf` instead of Regex, right?